### ISSUE TITLE
Create Shared Library for Public and Partners

### ISSUE THREAD

**Issue by seanmalbert** [bloom-housing/bloom#1625]
From the issue referenced below, UI Components is not the place to have functions that contain business logic. For this, we can create a separate package to be shared by Public and Partner sites.

It's definitely easiest to keep this logic contained within the single function. We could duplicate it across both partners and public but I'm not sure if that's a good idea either. Is it time to bite the bullet and create a "shared helpers" sort of package?

_Originally posted by @jaredcwhite in https://github.com/bloom-housing/bloom/pull/1568#discussion_r684491157_

**Comment by jaredcwhite** [comment]
Yeah this is something that's been discussed on and off again for ages…feels like the time has come￼. ⌛️

**Comment by jaredcwhite** [comment]
Current naming suggestion: `shared-helpers`, `site-helpers`

**Comment by jaredcwhite** [comment]
Based on an earlier conversation — the plan is we'll implement this package initially with just a few changes, and then handle additional work in future issues/PRs. That way we'll avoid dealing with a gigantic PR sporting hundreds of file changes.
